src/server/internalServer.cpp

```diff
@@ -119,6 +119,9 @@ Filter get_search_filter(const RequestContext& request, const std::string& prefi
     try {
       filter.rejectTags(kiwix::split(request.get_argument(prefix+"notag"), ";"));
     } catch (...) {}
+    try {
+      filter.aliasName(request.get_argument(prefix + "book"));
```

Comment:
My guess is that selection of books by alias/filename should not be combined with other filtering criteria. If so, then it rather must be done in `InternalServer::selectBooks()` (similar to selection by id or name).

Reply:
Sorry, I didn't understand this fully. We need a similar way to filter using query string (just like lang/category/tag)

Reply:
That was not the point of my comment. Does the following queries make sense?

- `catalog/v2/entries?book=<bookid>&lang=eng`
- `catalog/v2/entries?book=<bookid1>&book=<bookid2>&category=wikipedia`
- `catalog/v2/entries?book=<bookid>&tag=nopic`

Reply:
@veloman-yunkan Not sure to understand your remark either (is that a principle remark about filtering by humanid or a just an implementation remark). Your example don't include any "human_id" argument? A request of the type `catalog/v2/entries?book=<bookid>&lang=eng` can not fullfill the feature request because it is not stable over time. The goal here is to have a way to selection an arbitrary list of ZIM file over time. AFAIK, using the human ids and there corresponding aliases is the only way for the moment... and so far it has works fine for the Web URLS (for example library.kiwix.org/wikipedia_en_all_maxi/).

Reply:
@kelson42 In my examples `<bookid>` is supposed to be the "human_id". The examples illustrate the following cases:

- selection by "human_id" is combined with filtering by language
- selection by "human_id" is combined with filtering by category
- selection by "human_id" is combined with filtering by tag

In my understanding, selection by human_id should be mutually exclusive with other ways of content filtering.

Reply:
@veloman-yunkan I understand you now. I'm not in favour of hardwiring exclusivity but yes, your examples don't really make sense. For example, if the `humanid` is for a book in Armenian, then if `lang=eng` then not result, but if `lang=arm` then if will give one result. Etc...

Reply:
@kelson42 My vision is that human_id must not be combined with other selection criteria. My [original proposal](https://github.com/kiwix/libkiwix/pull/808#discussion_r954639462) applies only to such a use model.

Reply:
@kelson42 In other words selection by humanid should work similar to selection by name or uuid (in which case other filtering criteria are ignored).

Reply:
Nothing fondamental against this, even if I would prefer to avoid a special treatment. @juuz0 good to you to fix the points noticed by @veloman-yunkan ?